RealmSwift/RealmKeyedCollection.swift

```diff
@@ -0,0 +1,856 @@
+
+     - parameter keyPath:  The key path to sort by.
+     - parameter ascending: The direction to sort in.
+     */
+    func sorted(byKeyPath keyPath: String, ascending: Bool) -> Results<Value>
+
+    /**
+     Returns a `Results` containing the objects in the dictionary, but sorted.
+
+     - warning: Dictionaries may only be sorted by properties of boolean, `Date`, `NSDate`, single and double-precision
+                floating point, integer, and string types.
+
+     - see: `sorted(byKeyPath:ascending:)`
+    */
+    func sorted<S: Sequence>(by sortDescriptors: S) -> Results<Value>
+        where S.Iterator.Element == SortDescriptor
+
+    /// Returns all of the keys in this dictionary.
+    var keys: [Key] { get }
+
+    /// Returns all of the values in the dictionary.
+    var values: [Value] { get }
+
+    /**
+     Returns the object at the given `position`.
+
+     - parameter position: The position of the element in the collection.
+     */
+    subscript(position: MapIndex) -> (key: Key, value: Value) { get }
+
+    /// Returns the position of an element in the collection.
+    /// - Parameter object: The object to find.
+    func index(of object: Value) -> MapIndex?
+
+    // MARK: Aggregate Operations
+
+    /**
+     Returns the minimum (lowest) value of the given property among all the objects in the collection, or `nil` if the
+     dictionary is empty.
+
+     - warning: Only a property whose type conforms to the `MinMaxType` protocol can be specified.
+
+     - parameter property: The name of a property whose minimum value is desired.
+     */
+    func min<T: MinMaxType>(ofProperty property: String) -> T?
+
+    /**
+     Returns the maximum (highest) value of the given property among all the objects in the dictionary, or `nil` if the
+     dictionary is empty.
+
+     - warning: Only a property whose type conforms to the `MinMaxType` protocol can be specified.
+
+     - parameter property: The name of a property whose minimum value is desired.
+     */
+    func max<T: MinMaxType>(ofProperty property: String) -> T?
+
+    /**
+    Returns the sum of the given property for objects in the dictionary, or `nil` if the dictionary is empty.
+
+    - warning: Only names of properties of a type conforming to the `AddableType` protocol can be used.
+
+    - parameter property: The name of a property conforming to `AddableType` to calculate sum on.
+    */
+    func sum<T: AddableType>(ofProperty property: String) -> T
+
+    /**
+     Returns the average value of a given property over all the objects in the dictionary, or `nil` if
+     the dictionary is empty.
+
+     - warning: Only a property whose type conforms to the `AddableType` protocol can be specified.
+
+     - parameter property: The name of a property whose values should be summed.
+     */
+    func average<T: AddableType>(ofProperty property: String) -> T?
+
+    // MARK: Notifications
+
+    /**
+     Registers a block to be called each time the dictionary changes.
```

Comment:
There's a bunch of wording changes needed here and the example isn't even using a dictionary.